### PR TITLE
Fix #868: Option to ignore globally-namespaced classes from MissingImport

### LINKED ISSUE #868: Exclude global namespace classes from MissingImport rule
## Description
The [MissingImport rule](https://github.com/phpmd/phpmd/blob/1964e0f81ceec3c2b15281d2709dbb66b68873c1/src/main/resources/rulesets/cleancode.xml#L185) flags any class that is referenced via its FQN. This makes sense for any class that is part of a namespace, but it's not so clear that this should apply to classes in the global namespace (e.g. `Exception`, `RuntimeException`).

This isn't just an aesthetic annoyance, it makes it impossible to use this PHPMD rule on some types of projects (e.g. Drupal) that specifically enforce that classes in the global namespace be fully qualified:
- https://www.drupal.org/docs/develop/coding-standards/namespaces#s-use-ing-classes
- https://www.drupal.org/project/coder/issues/3120001#comment-13993942

For instance, this passes Drupal coding standards:
```php
<?php

namespace Foo;

throw new \Exception;
```

This passes PHPMD:
```php
<?php

namespace Foo;

use \Exception;

throw new Exception;
```

But there's no way that I can tell to make it work with both standards.

Can we exclude global namespace classes from this rule (either by default or as an option)?

I _might_ be able to submit a PR if you approve, I'm not sure if rules can have options.

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [x] Tell if you have the option to provide the code for this proposal.
* [x] This issue is about 1 feature proposal and nothing more.
* [x] The issue has a descriptive title. For example: "Add JSON render option".

### PR DESCRIPTION
**Pull request by danepowell**
Type: feature
Issue: Resolves #868
Breaking change: no

See #868 for motivation. Adds a property `ignore-global` to the MissingImport rule that will cause it to ignore classes in the global namespace (i.e. [classes that do not contain a backslash in their FQN](https://www.drupal.org/project/coder/issues/3120001)).

This still needs tests but I'd like maintainer approval and guidance before proceeding with that, since I'm a little intimidated by the unit test architecture. I don't know how to incorporate an optional flag into the test cases.

As an aside, kudos on maintaining a very well-organized and documented codebase, it was pretty easy to grok.

### CODE REVIEW

**Comment by tvbeek** [comment]
@danepowell thanks for the PR. Maybe you already got an notification / email from the GitHub actions but there are failing test. If I take a quick look it is the result of not having a default value. Can you check check and update it? (Not having a default value that keeps the current logic should also make it a breaking change)

**Comment by danepowell** [comment]
Thanks, I fixed the broken test and also added a test case for the new option.

File: src/site/rst/rules/cleancode.rst [hunk 1]
```diff
@@ -132,6 +132,14 @@ Example: ::
       return new \stdClass();
   }
 
+This rule has the following properties:
+
++-----------------------------------+---------------+------------------------------------------------------------+
+| Name                              | Default Value | Description                                                |
++===================================+===============+============================================================+
+| ignore-global                     |               | Ignore classes in the global namespace                     |
```
Comment (kylekatarnls) [inline src/site/rst/rules/cleancode.rst:140]:
```suggestion
| ignore-global                     |               | Ignore classes and interfaces in the global namespace      |
```

File: src/site/rst/rules/cleancode.rst [hunk 2]
```diff
@@ -132,6 +132,14 @@ Example: ::
       return new \stdClass();
   }
 
+This rule has the following properties:
+
++-----------------------------------+---------------+------------------------------------------------------------+
+| Name                              | Default Value | Description                                                |
++===================================+===============+============================================================+
+| ignore-global                     |               | Ignore classes and interfaces in the global namespace      |
```
Comment (tvbeek) [inline src/site/rst/rules/cleancode.rst:140]:
```suggestion
| ignore-global                     |               | Ignore classes, interfaces and traits in the global namespace      |
```

I think after the comment from @kylekatarnls that it is good to make the description as complete as possible.

**Review by tvbeek — CHANGES_REQUESTED** [review]
I have a small request for change, just to make it more explicit. After that I think it is perfect.

**Review by tvbeek — APPROVED** [review]

**Comment by tvbeek** [comment]
Thanks for the PR

**Pull request merged**